tests/fv/fv_test.go

```diff
@@ -81,6 +81,101 @@ var _ = Describe("PolicyController", func() {
 		apiserver.Stop()
 	})
 
+	Context("nodes", func() {
+		// It("should delete a node that exists at start of day", func() {})
+		// It("should delete nodes that nodeRef unexisting node in k8s"), func() {})
+		// It("should clean up dead data for nodes"
+		It("should be removed in response to a k8s node delete", func() {
+			kn := &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "k8snode",
+				},
+			}
+			_, err := k8sClient.CoreV1().Nodes().Create(kn)
+			Expect(err).NotTo(HaveOccurred())
+
+			cn := &api.Node{
+				Metadata: api.NodeMetadata{
+					Name: "calnode",
+				},
+				Spec: api.NodeSpec{
+					OrchRefs: []api.OrchRef{
+						api.OrchRef{
+							NodeName: "k8snode",
+							Orchestrator: "k8s",
+						},
+					},
+				},
+			}
+			_, err = calicoClient.Nodes().Create(cn)
+			Expect(err).NotTo(HaveOccurred())
+
+			k8sClient.CoreV1().Nodes().Delete("k8snode", &metav1.DeleteOptions{})
+			Eventually(func() *api.Node {
+				node, _ := calicoClient.Nodes().Get(api.NodeMetadata{Name: "calnode"})
+				return node
+			}, time.Second*15, 500*time.Millisecond).Should(BeNil())
+		})
+
+		It("should be removed if they reference a k8sNode that doesn't exist", func() {
+			cn := &api.Node{
+				Metadata: api.NodeMetadata{
+					Name: "calnode",
+				},
+				Spec: api.NodeSpec{
+					OrchRefs: []api.OrchRef{
+						{
+							NodeName: "k8sfakenode",
+							Orchestrator: "k8s",
+						},
+					},
+				},
+			}
+			_, err := calicoClient.Nodes().Create(cn)
+			Expect(err).NotTo(HaveOccurred())
+
+			k8sClient.CoreV1().Nodes().Delete("k8snode", &metav1.DeleteOptions{})
+			Eventually(func() *api.Node {
+				node, _ := calicoClient.Nodes().Get(api.NodeMetadata{Name: "calnode"})
+				return node
+			}, time.Second*15, 500*time.Millisecond).Should(BeNil())
+		})
+
+		It("should not be removed if they don't have any orchrefs.", func() {
+			cn := &api.Node{
+				Metadata: api.NodeMetadata{
+					Name: "calnode",
+				},
+				Spec: api.NodeSpec{
+					OrchRefs: []api.OrchRef{},
```

Comment:
What about an OrchRef from another orchestrator? That might be a more useful test than this one.